lib/compiler/src/v3_kernel.erl

```diff
@@ -2109,6 +2108,48 @@ new_clauses(Cs0, U, St) ->
 	      end, Cs0),
     {Cs1,St}.
 
+%% count and squeeze
+%%  The goal of those functions is to count and squeeze subsequent literal
+%%  integer k_bin_seg so we can leverage bs_get_integer_16 and similar.
+%%  Note the optimization still applies if the last clause is a k_bin_end
+%%  or a variadic segment with size equal to or less than the previous sizes.
+count_clauses_bin_integer_segments([#iclause{pats=[#k_bin_seg{seg=#k_literal{}} = BinSeg]} | Clauses], Min) ->
+    NewMin = min(count_bin_integer_segments(BinSeg, 0), Min),
+    count_clauses_bin_integer_segments(Clauses, NewMin);
+count_clauses_bin_integer_segments([#iclause{pats=[#k_bin_seg{size=#k_int{val=Size},unit=Unit,
+	    type=integer,flags=[unsigned,big], seg=#k_var{}}]}], Min)
+	    when ((Size * Unit) rem 8) =:= 0, ((Size * Unit) div 8) =< Min ->
+    (Size * Unit) div 8;
+count_clauses_bin_integer_segments([#iclause{pats=[#k_bin_end{}]}], Min) ->
+    Min;
+count_clauses_bin_integer_segments([], Min) ->
+    Min;
+count_clauses_bin_integer_segments(_, _) ->
+    0.
+
+count_bin_integer_segments(_, ?SQUEEZE_MAX_BYTES_SEGMENT) ->
+    ?SQUEEZE_MAX_BYTES_SEGMENT;
+count_bin_integer_segments(#k_bin_seg{size=#k_int{val=8},unit=1,type=integer,flags=[unsigned,big],
+	    seg=#k_literal{val=Val},next=Next}, Count) when is_integer(Val) ->
+    count_bin_integer_segments(Next, Count + 1);
+count_bin_integer_segments(_, Count) ->
+    Count.
+
+squeeze_clauses_bin_integer_segments(Clauses, Count) when Count > 1 ->
+    lists:map(fun
```

Comment:
In the compiler, we import functions from the `lists` module instead of writing a `lists:` prefix.